xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ConfigurableClassMacros.xml

```diff
@@ -85,34 +85,25 @@
  *###
 #macro(findNamesOfAppsToConfigure, $section, $globaladmin, $space, $outputList)
   ## Use a parameterized sql query to prevent injection.
-  #set($params = [])
-  #if($section &amp;&amp; $section != '')
-    #set($discard = $params.add("$section"))
-    #set($sqlA = ' StringProperty as section,')
-    #set($sqlB = " and section.id=obj.id and section.name='displayInSection' and section.value=?")
+  #set ($params = [])
+  #set ($from = ', BaseObject as obj, IntegerProperty as global, StringProperty as visibility')
+  #set ($where = "where doc.fullName=obj.name and obj.className='$nameOfThisDocument' and ((obj.id=global.id.id and global.id.name='configureGlobally' and global.value=?) or (obj.id=visibility.id.id and visibility.id.name='visibility' and visibility.value in ('global',?)))")
```

Comment:
is this going to work? AFAIK $nameOfThisDocument is not going to be evaluated since it's inside single quotes, no?